source/adapters/level_zero/context.hpp

```diff
@@ -142,6 +142,9 @@ struct ur_context_handle_t_ : _ur_object {
   //
   // Cache of event pools to which host-visible events are added to.
   std::vector<std::list<ze_event_pool_handle_t>> ZeEventPoolCache{4};
+  std::vector<std::unordered_map<ze_device_handle_t,
+                                 std::list<ze_event_pool_handle_t> *>>
+      ZeEventPoolCacheDeviceMap{4};
```

Comment:
why 4 here as the default value? I know we have that above, I am just curious if there is a particular reason to init with 4.

Reply:
@nrspruit : It's to have a different event pool cache for each combination of scope (Host visibility) and profiling mode, as these two traits are used when originally creating an Event pool in the L0 URT.